In [22]:
import pandas as pd
import os as os

In [12]:
data = pd.read_csv('OnlineRetail.csv', encoding='ISO-8859-1')

In [13]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  object 
 1   StockCode    541909 non-null  object 
 2   Description  540455 non-null  object 
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  object 
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 33.1+ MB


In [23]:
data.describe()

Quantity
count  541909.000000
mean        9.552250
std       218.081158
min    -80995.000000
25%         1.000000
50%         3.000000
75%        10.000000
max     80995.000000

Nhận xét về dữ liệu:
- Cấu trúc:
  + Bộ dữ liệu có 8 cột và 541909 dòng
  + Các cột InvoiceNo, StockCode, Description, Country thuộc kiểu dữ liệu object
  + Các cột UnitPrice, CustomerID thuộc kiểu dữ liệu float64
  + Cột Quantity thuộc kiễu dữ liệu int64
  * Thuộc tính, thước đo: 
    + InvoiceNo, StockCode, Description, Country là thuốc tính định tính, có thước đo định danh
    + CustomerID tuy mang kiểu dữ liệu là float64 nhưng nó dùng để phân biệt giữ các thực thể khác nhau nên mang thuộc tính định danh và thước đo định danh
    + Quantity, UnitPrice mang thuộc tính định lượng và thước đo ratio
- Phân bố dữ liệu:
  + Cột Quantity:
    * Giá trị trung bình: ~ 9.55
    * Giá trị độ lệch chuẩn: ~ 218.08 lệch xa giá trị trung bình
    * Giá trị min: -80995 đối với cột số lượng có số âm là không hợp lý.
    * Giá trị max: 80995 lệch xa với giá trị độ lệch chuẩn và giá trị trung bình.
    * 25% số lượng đặt hàng <1 
    * 50% số lượng đặt hàng <3
    * 75% số lượng đặt hàng <10
    => Q1, Q2, Q3 lệch không đáng kể so với giá trị trung bình, nhưng lệch khá xa so với độ lệch chuẩn
  + Cột UnitPrice
    * Giá trị trung bình: ~ 4.61
    * Giá trị độ lệch chuẩn: ~ 96.76 lệch xa giá trị trung bình
    * Giá trị min: -11062.060000 đối với cột đơn giá có số âm là không hợp lý.
    * Giá trị max: 38970 lệch xa với giá trị độ lệch chuẩn và giá trị trung bình.
    * 25% số lượng đặt hàng <1.25 
    * 50% số lượng đặt hàng <2.08
    * 75% số lượng đặt hàng <4.13
    => Q1, Q2, Q3 lệch không đáng kể so với giá trị trung bình, nhưng lệch khá xa so với độ lệch chuẩn 

In [24]:
data = data[['Description','Quantity']]

In [25]:
data

Description  Quantity
0        WHITE HANGING HEART T-LIGHT HOLDER         6
1                       WHITE METAL LANTERN         6
2            CREAM CUPID HEARTS COAT HANGER         8
3       KNITTED UNION FLAG HOT WATER BOTTLE         6
4            RED WOOLLY HOTTIE WHITE HEART.         6
...                                     ...       ...
541904          PACK OF 20 SPACEBOY NAPKINS        12
541905         CHILDREN'S APRON DOLLY GIRL          6
541906        CHILDRENS CUTLERY DOLLY GIRL          4
541907      CHILDRENS CUTLERY CIRCUS PARADE         4
541908        BAKING SET 9 PIECE RETROSPOT          3

[541909 rows x 2 columns]

In [26]:
df_data = data.head(1000)

In [27]:
df_data

Description  Quantity
0     WHITE HANGING HEART T-LIGHT HOLDER         6
1                    WHITE METAL LANTERN         6
2         CREAM CUPID HEARTS COAT HANGER         8
3    KNITTED UNION FLAG HOT WATER BOTTLE         6
4         RED WOOLLY HOTTIE WHITE HEART.         6
..                                   ...       ...
995                HEART OF WICKER SMALL         1
996             SKULLS SQUARE TISSUE BOX         1
997      PINK PAISLEY SQUARE TISSUE BOX          1
998         PACK OF 6 HANDBAG GIFT BOXES         1
999           TOAST ITS - HAPPY BIRTHDAY         2

[1000 rows x 2 columns]

In [28]:
os.makedirs('Data_Excel', exist_ok=True)  


In [30]:
df_data.to_excel('Data_Excel\OnlineRetail.xlsx', index=False)  
print('Save to Excel file successfully.')

Save to Excel file successfully.
